kubernetes-client/src/main/java/io/fabric8/kubernetes/client/CustomResource.java

```diff
@@ -80,4 +129,120 @@ public ObjectMeta getMetadata() {
   public void setMetadata(ObjectMeta metadata) {
     this.metadata = metadata;
   }
+  
+  /**
+   * Retrieves the plural form associated with the specified CustomResource if annotated with {@link Plural} or computes a default value
+   * using the value returned by {@link #getSingular(Class)} as input to {@link Pluralize#toPlural(String)}.
+   *
+   * @param clazz the CustomResource whose plural form we want to retrieve
+   * @return the plural form defined by the {@link Plural} annotation or a computed default value
+   */
+  public static String getPlural(Class<? extends CustomResource> clazz) {
+    final Plural fromAnnotation = clazz.getAnnotation(Plural.class);
+    return (fromAnnotation != null ? fromAnnotation.value().toLowerCase(Locale.ROOT) : Pluralize.toPlural(getSingular(clazz)));
+  }
+  
+  @JsonIgnore
+  public String getPlural() {
+    if (plural == null) {
+      this.plural = getPlural(getClass());
+    }
+    return plural;
+  }
```

Comment:
I would rather see this `if` being part of the constructor and not the getter.

Reply:
I tried that but that broke deserialization, iirc. Then again, some other things have changed in the mean time so let me try that change again to see if it works now.

Reply:
Done.